python_version/keepassxc-proxy

```diff
@@ -12,108 +12,142 @@ import threading
-      if queue:
-        queue.put(None)
-      sys.exit(0)
-
-    text_length = struct.unpack('i', text_length_bytes)[0]
-    text = sys.stdin.read(text_length).decode('utf-8')
-
-    if queue:
-      queue.put(text)
-
-
-class NativeMessagingWindow(Tkinter.Frame):
-  def __init__(self, queue):
-    self.queue = queue
-
-    Tkinter.Frame.__init__(self)
-    self.pack()
-
-    self.text = Tkinter.Text(self)
-    self.text.configure(font=("Courier New", 12))
-    self.text.grid(row=0, column=0, padx=10, pady=10, columnspan=2)
-    self.text.config(state=Tkinter.DISABLED, height=20, width=80)
-    self.messageContent = Tkinter.StringVar()
-    self.after(100, self.processMessages) 
-
-  def processMessages(self):
-    while not self.queue.empty():
-      message = self.queue.get_nowait()
-      if message == None:
-        self.quit()
-        return
-      self.log("Received %s" % message)
-
-      # Handle requests from keepassxc-browser.
-      messageJSON = json.loads(message)
-      if messageJSON['action']:
-        if messageJSON['action'] == "change-public-keys" and messageJSON['proxyPort']:
-          self.log("Proxy port changed")
-          #serverPort = messageJSON['proxyPort']
-
-        udpSocket.sendto(message, ('localhost', serverPort))
-        try:
-          resp, server = udpSocket.recvfrom(4096)
-          rawResp = "Response: {}".format(resp)
-          self.log("Response: {}".format(rawResp))
-          try:
-            send_message(resp)
-          except IOError:
-            self.log("Failed to send message.")
-        except socket.timeout:
-          self.log("Receive timeout")
-          msg = json.dumps({"action": messageJSON['action'], "error": "Not connected with KeePassXC.", "errorCode": 5})
-          send_message(msg)
-
-    self.after(10, self.processMessages)
-
-  def log(self, message):
-    self.text.config(state=Tkinter.NORMAL)
-    self.text.insert(Tkinter.END, message + "\n")
-    self.text.config(state=Tkinter.DISABLED)
-
+    message_number = 0
+    while 1:
+        text_length_bytes = sys.stdin.read(4)
+
+        if len(text_length_bytes) == 0:
+            if queue:
+                queue.put(None)
+            sys.exit(0)
+
+        text_length = struct.unpack('i', text_length_bytes)[0]
+        text = sys.stdin.read(text_length).decode('utf-8')
+
+        if queue:
+            queue.put(text)
+
+class NativeMessaging():
+    def __init__(self, queue):
+        self.queue = queue
+
+    def processMessages(self):
+        while not self.queue.empty():
+            message = self.queue.get_nowait()
+            if message == None:
+                self.quit()
+                return
+            self.log("Received %s" % message)
+
+            # Handle requests from keepassxc-browser.
+            messageJSON = json.loads(message)
+            if messageJSON['action']:
+                sock.send(message)
+                try:
+                    resp, server = sock.recvfrom(4096)
```

Comment:
In the next version the maximum data size is 1024*1024. Please change this hard-coded value to a const variable.